src/libraries/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -657,15 +657,6 @@ void ReadRowOrCol(int startExclusive, int endExclusive, StdInReader reader, Read
                         result = row - 1;
                     }
                 }
-
-                void TransferBytes(ReadOnlySpan<byte> src, StdInReader dst)
-                {
-                    for (int i = 0; i < src.Length; i++)
-                    {
-                        byte b = src[i];
-                        dst.AppendExtraBuffer(&b, 1);
```

Comment:
I'm not sure why the old logic would append the buffer byte-by-byte, but the new changes mean that this is no longer the case. Would this have any impact w.r.t. character decoding?

Reply:
It's possible, since it's using a stateless Encoding rather than a stateful Encoder.  Though that would seem to be something to be fixed regardless?  If you want me to back this out (or close the PR entirely) and investigate / address this separately, that's fine.

Reply:
It would seem it's by design, possibly the [commit that introduced it might contain a few hints](https://github.com/dotnet/corefx/commit/3aac19549465452c36f57ed3749ac8ee73e1956b#diff-cb11c7a5ca3a475a8382bf2353b2c115R395)? It would probably make sense to retain the `TransferBytes` method as-is, since it doesn't impact the goal of this PR.